src/dune/dune_file.ml

```diff
@@ -1565,6 +1571,10 @@ module Executables = struct
                      "See https://github.com/ocaml/dune/issues/745 for more \
                       details."
                  ]))
+    and+ statically_linked_libraries =
+      field_o "statically_linked_libraries"
```

Comment:
Why do we need this field? Why can't we just statically link all the libraries in `libraries`.

Reply:
The precise reason is that the plugin may require a library which is also linked into the main program. If you link the library into the plugin, then you won't be able to dynlink the plugin into the program because the modules of the library are already loaded. This is fairly common: 1) the common library may contains functionality that is used both by the main program and the plugins, or 2) it could be small utility libraries (think `uutf`) that happen to be used by both the main application and the plugins, etc.

Reply:
Even more concretely, the typical way plugins work is that they "register" themselves with the main program via state living on some shared module. This shared module is linked to the main program and should not/could not be linked into the plugins.